### PR TITLE
add provider and module versions for aws

### PR DESCRIPTION
**Pull request by DanielZhangQD**
### What problem does this PR solve? 
Some provider and module versions are missed in terraform scripts for AWS, which may cause new versions with breaking changes will be automatically installed by terraform init in future.
Refer to https://github.com/pingcap/tidb-operator/issues/895
### What is changed and how does it work?
- add provider and module versions for aws
- fix some typos
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    Deploy TiDB with terraform on AWS EKS

Code changes

 - Has terraform scripts change
 - Has documents change

Related changes

 - Need to cherry-pick to the release branch
    - cherry-pick-1.0
    - cherry-pick-1.1
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
NONE
 ```

### CODE REVIEW

File: deploy/modules/aws/tidb-operator/main.tf [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module "eks" {
   source = "terraform-aws-modules/eks/aws"
 
+  version            = "5.1.0"
```
Comment (cofyc) [inline deploy/modules/aws/tidb-operator/main.tf]: what is this version?

Comment (DanielZhangQD) [reply deploy/modules/aws/tidb-operator/main.tf]: This is the version for module terraform-aws-modules/eks/aws.
Each provider and the non-local module is recommended to set the version explicitly: 
https://www.terraform.io/docs/configuration/modules.html#module-versions
https://www.terraform.io/docs/configuration/providers.html#version-provider-versions

**Pull request merged**